transaction/extra/src/tx_summary_unblinding/verifier.rs

```diff
@@ -313,7 +313,11 @@ impl TxSummaryStreamingVerifier {
         mut self,
         fee: Amount,
         tombstone_block: u64,
-    ) -> ([u8; 32], TxSummaryUnblindingReport) {
+    ) -> Result<([u8; 32], TxSummaryUnblindingReport), Error> {
+        if self.input_count != self.expected_num_inputs {
```

Comment:
I know its implied that if the input count matches the number of inputs then we got all the outputs, but it might be worth adding that check anyway if someone calls finalize before they finished submitting the outputs.